core/utils/utils.go

```diff
@@ -819,7 +822,8 @@ var (
 // StartStopOnce contains a StartStopOnceState integer
 type StartStopOnce struct {
 	state        atomic.Int32
-	sync.RWMutex // lock is held during startup/shutdown, RLock is held while executing functions dependent on a particular state
+	sync.RWMutex             // lock is held during startup/shutdown, RLock is held while executing functions dependent on a particular state
+	Errbuffer    ErrorBuffer // errorbuffer tracks latest N crits in the service.
```

Comment:
consider change the comment indicate that service owners should use the errbuffer for tracking errors affecting the health report. maybe even rename to HealthReportErrBuffer or HealthReportErrs

As it reads right now, it implies that the errbuffer alone is sufficient to track the crits without any extra code.